### PR TITLE
Update the tool connecting to Pro and Earth

### PR DESCRIPTION
**Pull request by gis2all**
Add these functions into the tool:
- Add layer
- Removelayer
- Clear ArcGIS Earth workspace
- Save screenshot from ArcGIS Earth

### CODE REVIEW

File: ToArcGISEarth/ToArcGISEarth/AddLayerCheckBox.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                        if (layer.MapLayerType == ArcGIS.Core.CIM.MapLayerType.Operational)
+                        {
+                            addLayerJson["target"] = "OperationalLayers";
+                        }
+                        if (layer.MapLayerType == ArcGIS.Core.CIM.MapLayerType.BasemapBackground)
+                        {
+                            addLayerJson["target"] = "BasemapLayers";
+                        }
+                        string currentJson = addLayerJson.ToString();
+                        if (currentJson != null)
```
Comment (bentan2013) [inline ToArcGISEarth/ToArcGISEarth/AddLayerCheckBox.cs:59]: If you don't want output string cotain formatting. You can use ToString(Formatting.None)

File: ToArcGISEarth/ToArcGISEarth/AddLayerCheckBox.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        private void AddLayer(LayerEventsArgs args)
+        {
+            try
+            {
+                List<Layer> layerList = args.Layers as List<Layer>;
+                if (layerList != null && layerList.Count != 0)
+                {
+                    Layer layer = layerList[0] as Layer;
+                    string url = GetLayerUrl(layer);
+                    if (url != null)
```
Comment (bentan2013) [inline ToArcGISEarth/ToArcGISEarth/AddLayerCheckBox.cs]: !String.IsNullorEmpty will be better here

File: ToArcGISEarth/ToArcGISEarth/ClearArcGISEarthWorkspaceButton.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            else
+            {
+                this.IsChecked = false;
+                return;
+            }
+        }
+
+        private void ClearAll()
+        {
+            ConnectToArcGISEarthButton.Utils.ClearLayers("{\"target\":\"OperationalLayers\"}");
```
Comment (bentan2013) [inline ToArcGISEarth/ToArcGISEarth/ClearArcGISEarthWorkspaceButton.cs]: Just for information

you can also use {target: AllLayers}

Note, those won't clear bookmark.

File: ToArcGISEarth/ToArcGISEarth/ToArcGISEarth.csproj [hunk 4]
```diff
@@ -95,30 +95,22 @@
...
-  </ItemGroup>
-  <ItemGroup>
-    <AddInContent Include="DarkImages\GenericButtonBlue32.png" />
+    <Compile Include="RemoveLayerCheckBox.cs" />
+    <Compile Include="SaveImageFromArcGISEarthButton.cs" />
+    <Compile Include="SyncToArcGISEarthCheckBox.cs" />
+    <Compile Include="ConnectToArcGISEarthButton.cs" />
   </ItemGroup>
   <ItemGroup>
     <AddInContent Include="Images\ArcGISEarth16.png" />
```
Comment (bentan2013) [inline ToArcGISEarth/ToArcGISEarth/ToArcGISEarth.csproj:116]: Not sure we need change this name to ArcGISEarth18.png or just ArcGISEarth.png

**Review by bentan2013 — APPROVED** [review]
looks good to me.
I also write a few suggestions, you can have a look

File: ToArcGISEarth/ToArcGISEarth/ToArcGISEarth.csproj (hunk 4 above)
Comment (gis2all) [reply ToArcGISEarth/ToArcGISEarth/ToArcGISEarth.csproj:116]: No, ArcGISEarth16.png just represents image size, there also have ArcGISEarth32.png.

**Comment by gis2all** [comment]
Update, close the PR.

**Pull request merged**